lisa/transformers/rpm_kernel_installer.py

```diff
@@ -43,27 +52,28 @@ def validate(self) -> None:
                 f"The '{self.type_name()}' installer only support RPM based Distros. ",
             )
         runbook: RPMInstallerSchema = self.runbook
-        kernel_rpm_path: str = runbook.kernel_rpm_path
-        err: str = f"can not find kernel rpm file: {kernel_rpm_path}"
-        assert os.path.exists(kernel_rpm_path), err
-        assert kernel_rpm_path.endswith(
-            ".rpm"
-        ), f"Provided file {kernel_rpm_path} is not an rpm"
+        rpm_list = runbook.rpm_list
+        for rpm_path in rpm_list:
+            err: str = f"can not find kernel rpm file: {rpm_path}"
+            assert os.path.exists(rpm_path), err
+            assert rpm_path.endswith(".rpm"), f"Provided file {rpm_path} is not an rpm"
 
     def install(self) -> str:
         node = self._node
         runbook: RPMInstallerSchema = self.runbook
-        kernel_rpm_path: str = runbook.kernel_rpm_path
 
-        filename = os.path.basename(kernel_rpm_path)
-        remote_path = node.working_path / filename
-        node.shell.copy(
-            PurePath(kernel_rpm_path),
-            remote_path,
-        )
+        time_now = datetime.utcnow().strftime("%Y_%m_%d_%H_%M_%S")
+        # append time to folder name just in case we have multiple kernel installers
+        remote_path = node.working_path / f"kernel_rpms_{time_now}"
+        node.tools[Mkdir].create_directory(str(remote_path))
+        copy_tool = node.tools[RemoteCopy]
+        for rpm_file in runbook.rpm_list:
+            copy_tool.copy_to_remote(
```

Comment:
The copy should be done by the copy transformer, don't hard code it here. In some cases, it may need to support local testing.

Reply:
This copy is different from the copy transformer. This doesn't target a user given destination. I used `copy_to_remote` instead of `scp` tool, just because of its easy to use nature.

Reply:
What's mean a "user given destination"? The "copy transformer" is `FileUploaderTransformer`. The destination is defined in the transformer, it's the middle transition location to make the whole things works. The logic you implemented is similar with `FileUploaderTransformer`, but it assumes the rpm kernel installation only work, if the rpm files are locally. The assumption shouldn't be here to prevent other scenarios, like install by rpm in build, or test with local rpm packages.

Reply:
I meant there's no specific destination that needs to be taken through runbook. This is just an scp of the given local files to a remote directory under `node.working_path`. 

> but it assumes the rpm kernel installation only work, if the rpm files are locally. The assumption shouldn't be here to prevent other scenarios, like install by rpm in build, or test with local rpm packages.

Yes, I had this assumption to address the immediate use case. But since there is also `FileUploaderTransformer` which can take care of copying the files, Do you suggest continuing with this assumption of local file or change the assumption to remote file? We could also add a parameter `is_remote` to explicitly set through runbook. I want to maintain some uniformity and use same approach for the new `PackageInstallationTransformer` as well.

Reply:
the rpm installer should install only local files. If some files need to be copied to the tested node, it should be done by the file uploader transformer. It needs to minimize the functionality of each transformers, so they can be composted together for various scenario.
Sorry, I haven't found it assumes the packages are remote. It should be changed to use file uploader and local only rpm, and hope it doesn't bring too much work to your pipelines.

Reply:
Ok, got it. Then I will change the current Kernel RPMInstaller to use the path as local to the node (remote). And as suggested, FileUploaderTransformer can be used to copy the files. Will follow same approach for new PackageInstallationTransformer.
No, this doesn't bring much work in the pipelines. Thanks for the inputs.